src/controller/TypedReadCallback.h

```diff
@@ -149,6 +149,8 @@ class TypedReadEventCallback final : public app::ReadClient::Callback
         CHIP_ERROR err = CHIP_NO_ERROR;
         DecodableEventType value;
 
+        VerifyOrExit(apStatus != nullptr && !apStatus->IsSuccess(), err = apStatus->ToChipError());
```

Comment:
This will crash if `apStatus` is null.

Reply:
The condition is wrong, it should be negatived -- `apStatus == nullptr || apStatus->IsSuccess()`